hail/python/hail/vds/methods.py

```diff
@@ -34,14 +34,15 @@ def to_dense_mt(vds: 'VariantDataset') -> 'MatrixTable':
     :class:`.MatrixTable`
         Dataset in dense MatrixTable representation.
     """
+
+    # NOTE: There is a strong assumption that ref block LEN does not extend
+    # past the end of a contig. That is bad data and we won't correct for it.
+    # Garbage in, garbage out.
     ref = vds.reference_data
-    # FIXME(chrisvittal) consider changing END semantics on VDS to make this better
-    # see https://github.com/hail-is/hail/issues/13183 for why this is here and more discussion
-    # we assume that END <= contig.length
-    ref = ref.annotate_rows(_locus_global_pos=ref.locus.global_position(), _locus_pos=ref.locus.position)
-    ref = ref.transmute_entries(_END_GLOBAL=ref._locus_global_pos + (ref.END - ref._locus_pos))
+    ref = ref.annotate_rows(_locus_global_pos=ref.locus.global_position())
+    ref = ref.transmute_entries(_END_GLOBAL=ref._locus_global_pos + ref.LEN)
```

Comment:
Can you remind me the mapping between `end` and `len`? I thought `end` was inclusive, so that `len = end - pos + 1`.

Reply:
This is correct. Because `_END_GLOBAL` is a temporary, I didn't think much of changing it to be an open interval rather than a closed one, but for the purpose of clarity, I think I'll change it back.